### PR TITLE
Dev/issue 8894 rest api v3

### PR DESCRIPTION
**Pull request by mcantelon**
METS parsing work and REST server.

### CODE REVIEW

**Comment by Hwesta** [comment]
None of the PREMIS tests pass - please fix

File: metsrw/metadata.py [hunk 1]
```diff
@@ -299,11 +356,17 @@ def parse(cls, root):
         mdtype = root.get('MDTYPE')
         if not mdtype:
             raise exceptions.ParseError('mdWrap must have a MDTYPE')
+        if mdtype in MDWrap.MDTYPE_CLASSES.keys():
+            mdtype_class = MDWrap.MDTYPE_CLASSES[mdtype]()
+            data = mdtype_class.parse(root.find('mets:xmlData', namespaces=utils.NAMESPACES)).__dict__
```
Comment (Hwesta) [inline metsrw/metadata.py]: Why store the `__dict__` of this instead of the parsed class itself?

File: metsrw/premis/event.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            raise exceptions.ParseError('Object can only parse event elements with PREMIS namespace.')
+
+        identifier_el = root.find("premis:eventIdentifier", namespaces=utils.NAMESPACES)
+
+        if identifier_el is not None:
+            identifier_type = identifier_el.findtext("premis:eventIdentifierType", namespaces=utils.NAMESPACES)
+            identifier_value = identifier_el.findtext("premis:eventIdentifierValue", namespaces=utils.NAMESPACES)
+        else:
+            identifier_type = None
+            identifier_value = None
```
Comment (Hwesta) [inline metsrw/premis/event.py]: If these are mandatory in the constructor, why allow them to be set to None here?

File: metsrw/premis/event.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                'roles': roles
+            })
+
+        return identifiers
+
+    def serialize(self):
+        root = self._document_root()
+        root.append(self._serialize_identifier())
+
+        if self.event_type is not None:
```
Comment (Hwesta) [inline metsrw/premis/event.py]: This and other elements can't be none - these checks are necessary.

File: metsrw/metadata.py (hunk 1 above)
Comment (mcantelon) [inline metsrw/metadata.py]: It makes it easier to convert the parsed data to JSON.

**Comment by Hwesta** [comment]
The REST server should be its own PR, since it's a candidate for being a separate repo, and quite different from the PREMIS parsing.

**Comment by mcantelon** [comment]
Closing this to separate the two features into two PRs.

**Pull request closed without merging**